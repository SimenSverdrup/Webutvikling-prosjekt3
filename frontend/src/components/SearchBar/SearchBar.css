#searchBar_grid_container {
    grid-column-start: 1;
    grid-column-end: 11;
    grid-row-start: 1;
    grid-row-end: 2;
}

.searchBar_grid {
    background-color: #C0C0C0;
    display: grid;
    grid-template-columns: 50% 25% 25%;
    border-radius: 5px;
}

input {
    grid-column-start: 1;
    grid-column-end: 2;
    margin: 7px;
    padding: 10px;
    border-radius: 4px;
    border-width: 1px;
}

#searchBar_sort {
    grid-column-start: 2;
    grid-column-end: 3;
    margin-left: 45%;
}

#searchBar_filter {
    grid-column-start: 3;
    grid-column-end: 4;
    margin-left: 30%;
}

