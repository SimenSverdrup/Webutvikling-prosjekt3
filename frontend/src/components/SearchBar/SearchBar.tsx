import React, {useContext, useState }from 'react';
import './SearchBar.css';
import Store from '../../mobx/store';
import MenuListComposition1 from "../../components/Dropdown/DropdownSort";
import LongMenu from "../../components/Dropdown/dropdownFilterSlider";


const SearchBar = () => {
    const store = useContext(Store);
    const { updateSearch } = store;

    return (
        <div id="searchBar_grid_container">
            <div className="searchBar_grid">
                <input className='searchInput' type="text" placeholder="Search" onInput={(element) => {
                    //use action here, to set search string state in the store
                    // this will trigger a re-render of MovieList automatically
                    updateSearch(element.currentTarget.value);
                    }}/>
                <div id="searchBar_sort" onClick={() => {/* P책 klikk vil jeg 책pne dropdown for sortering */}}>
                    <MenuListComposition1/>
                </div>
                <div id="searchBar_filter" onClick={() => {/* P책 klikk vil jeg 책pne dropdown for filtrering */}}>
                    <LongMenu/>
                </div>
            </div>
        </div>
    )
}

export default SearchBar;
