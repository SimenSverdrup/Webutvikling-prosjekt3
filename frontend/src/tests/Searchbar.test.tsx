import React from 'react';
import { render } from '@testing-library/react' // ES6
import SearchBar from '../components/SearchBar/SearchBar';



it("renders SearchBar correctly", () => {
        const { container } = render(<SearchBar/>);
        expect(container).toMatchSnapshot();
});

