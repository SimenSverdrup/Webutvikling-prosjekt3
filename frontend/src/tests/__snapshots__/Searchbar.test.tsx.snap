// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchBar correctly 1`] = `
<div>
  <div
    id="searchBar_grid_container"
  >
    <div
      class="searchBar_grid"
    >
      <input
        placeholder="Search"
        type="text"
      />
      <div
        id="searchBar_sort"
      >
        <div
          class="makeStyles-root-1"
        >
          <div>
            <button
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Sort:
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        id="searchBar_filter"
      >
        <div>
          <button
            aria-haspopup="true"
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Filter:
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
